tests/common/macros.h

```diff
@@ -56,18 +56,34 @@ inline void set_test_info(sycl_cts::util::test_base::info& out,
 }
 
 /**
- *
+ * Helper function to log details for SYCL exception
  */
 inline void log_exception(sycl_cts::util::logger& log,
                           const sycl::exception& e) {
-  // notify that an exception was thrown
-  log.note("sycl exception caught");
-
-  // log exception error string
-  std::string what = e.what();
-  if (!what.empty()) {
-    log.note("what - " + what);
-  }
+  std::string message = "SYCL exception\n";
+  // Using references to avoid object slicing
+  const auto& code = e.code();
+  const auto& category = e.category();
+
+  auto append_str = [&message](const char* description, std::string&& value) {
```

Comment:
```suggestion
  auto append_str = [&message](const char* description, const std::string& value) {
```

Reply:
Wouldn't it trigger unnecessary copy constructor in this case?

Reply:
Where ?

Reply:
Passing by `const` reference is cheaper here than the move construction. Also, using an r-value reference prevents the l-value usage like
```
std::string a { "hello" };
append_str("yo", a);
```

Reply:
Missed `std::move` here. The `std::basic_string::operator+` has [different overloads](https://en.cppreference.com/w/cpp/string/basic_string/operator%2B) for l-value and r-value, so r-value restriction was intended. 
Still looks like any possible performance gain here doesn't worth it in the terms of further support, so it was aligned with the codebase to use `const std::string&`. Not to mention that `operator+` is not used anymore )

Reply:
I am unsure about what you mean with "Missed `std::move` here." If this is about a missing `std::move` in my snippet, this is on purpose, since `a` could be used also later on.
But I think you are right, we do not care about these micro-optimziations in the CTS. :-)

Reply:
Oops, just to clarify - I meant that I had missed the `std::move` in original code. Sorry.